rfcs/43-many-worlds.md

```diff
@@ -0,0 +1,522 @@
+# Feature Name: `many-worlds`
+
+## Summary
+
+Bevy apps: now with more worlds. And more schedules to match!
+Transfer entities between worlds, clone schedules and spin up and down worlds with the `AppCommands`.
+New `GlobalRes` resource type, which is accessible across worlds.
+
+## Motivation
+
+The ability to have multiple worlds (and unique groups of systems that run on them) is a very useful, reusable building block for separating parts of the app that interact very weakly, if at all.
+Some of the more immediate use cases include:
+
+1. A more unified design for pipelined rendering.
+2. Entity staging grounds, where complete or nearly-complete entities can be stored for quick later use (see [bevy #1446](https://github.com/bevyengine/bevy/issues/1446)).
+3. Entity purgatories, where despawned entities can be kept around while their components are inspected (see [bevy #1655](https://github.com/bevyengine/bevy/issues/1655)).
+4. A trivially correct structure for parallel world simulation for scientific simulation use cases.
+5. Look-ahead simulation for AI, where an agent simulates a copy of the world using the systems without actually affecting game state.
+6. Isolation of map chunks.
+7. Temporarily disabling entities in a system-agnostic fashion.
+
+Currently, this flavor of design requires abandoning `bevy_app` completely, carefully limiting *every* system used (including base / external systems) or possibly some very advanced shenanigans with exclusive systems.
+
+A more limited "sub-world" design (see #16 for further discussion) is more feasible, by storing a `World` struct inside a resource, but this design is harder to reason about, unnecessarily hierarchical and inherently limited.
+
+## User-facing explanation
+
+When you need to fully isolate entities from each other (or make sure logic doesn't actually run on some chunk of them), you can create **multiple worlds** in your Bevy app.
+
+At this point, you should be familiar with the basics of both `Worlds` and `Schedules`.
+Your `App` can store any number of these, each corresponding to a particular `WorldLabel`.
+
+During each pass of the game loop, each schedule runs in parallel, modifying the world it is assigned to.
+These worlds are fully isolated: they cannot view or modify any other world, and stages progress independently across schedules.
+Then, the **global world** (which stores common, read-only data) is processed, and any systems in its schedule are run.
```

Comment:
I think global resources are needed. Something like `AssetServer` or `WgpuSettings` would belong there. Global entities are not a good idea I think, so maybe it doesn't need to be a `World`. A global schedule would be good tho. This would be the place to put systems that are exclusive over all worlds.